core/domain/stats_domain_test.py

```diff
@@ -138,3 +159,98 @@ def test_resolving_answers(self):
             'eid', state_name, self.DEFAULT_RULESPEC_STR)
         self.assertEquals(answer_log.answers, {'answer2': 1})
         self.assertEquals(answer_log.total_answer_count, 1)
+
+
+class StateAnswersTests(test_utils.GenericTestBase):
+    """Test the state answers domain object."""
+
+    DEFAULT_RULESPEC_STR = exp_domain.DEFAULT_RULESPEC_STR
+
+    def test_record_answer(self):
+        self.save_new_default_exploration('eid', 'fake@user.com')
+        exp = exp_services.get_exploration_by_id('eid')
+
+        FIRST_STATE_NAME = exp.init_state_name
+        SECOND_STATE_NAME = 'State 2'
+        exp_services.update_exploration('fake@user.com', 'eid', [{
+            'cmd': 'edit_state_property',
+            'state_name': FIRST_STATE_NAME,
+            'property_name': 'widget_id',
```

Comment:
I think it would be better to use exp_domain.STATE_PROPERTY_INTERACTION_ID instead of 'widget_id', to ensure there is one source of truth for this value.